collecte/pdf_downloader.py

```diff
@@ -31,60 +47,441 @@ def cbcr_finder(
             link = item["link"]
             if link.endswith(".pdf"):
                 pdf_urls.append(link)
+        logger.logger.debug(f"\nURLs found for company '{company_name}':")
+        for url in pdf_urls:
+            logger.logger.debug(f"{url}")
     else:
-        print("No results found.")
+        logger.logger.error(f"No results found for query '{search_query}'")
 
     return pdf_urls
 
 
-def download_pdf(url: str, download_folder: str, company_name: str) -> str:
+def download_pdf(
+    url: str,
+    download_folder: Path,
+    website_name: str,
+    fetch_timeout_s: int,
+    check_pdf_integrity: bool,
+) -> (str, None | Exception):
+
     # Create a sanitized version of the company name for the directory
-    company_folder = Path(download_folder) / "".join(
-        e for e in company_name if e.isalnum()
-    )
+    company_folder = download_folder / website_name
 
     Path.mkdir(company_folder, parents=True, exist_ok=True)
-
     local_filename = Path(company_folder) / url.split("/")[-1]
+    exception_status = None
 
     if not Path.exists(local_filename):
+
         try:
-            with requests.get(url, stream=True) as r:
+            t_start = time()
+            with requests.get(url, stream=True, timeout=(3.05, 10)) as r:
```

Comment:
For some companies like bhp the request times out but it can be fixed by changing the headers (to mimic a request from a browser:
```python
{
    "User-Agent": (
        "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 "
        "(KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.3"
    ),
}
```
I don't know if this solution is suitable though
)

Reply:
Yes, the request on `"https://www.bhp.com/-/media/documents/ourapproach/operatingwithintegrity/taxandtransparency/231221_countrybycountryreport2022.pdf"` times out without the header and downloads the file properly with it